first pass at adding a non-selectively disclosable JWT VC to the spec (#120)
vcstuff/draft-terbu-sd-jwt-vc: draft-terbu-sd-jwt-vc.md

%%%

> title = "Verifiable Credentials with a JWT Claim Set with or without Selective Disclosure"

Author: yeah, title.. `JWT based` and `with a JWT Claim set` is different in my mind because `JWT != SD-JWT`, and the most important thing we are trying to convey is that the payload is vanilla JSON and I found `with a JWT Claim set` to be a good compromise because JSON-LD payload does not use JWT claims while JSON-LD is still JSON.

> title = "Verifiable Credentials with a JWT Claim Set with or without Selective Disclosure"

Reviewer: I'd argue that a shorter more succinct title like `JWT and SD-JWT Verifiable Credentials` would be better. Any subtleties could be added/addressed in the abstract or intro. If needed. But the title doesn't seem like the right place to be differentiating a JWT claim set (which is JSON) or JSON or JSON-LD or whatever.

> title = "Verifiable Credentials with a JWT Claim Set with or without Selective Disclosure"

Reviewer: And yes, I know there's a recent [title change committed to SD-JWT](https://github.com/oauth-wg/oauth-selective-disclosure-jwt/commit/332ded8faf20d528c5c05e04cebed28aab1a0d33) but I don't love it.

> title = "Verifiable Credentials with a JWT Claim Set with or without Selective Disclosure"

Reviewer 2: > I'd argue that a shorter more succinct title like JWT and SD-JWT Verifiable Credentials would be better. Any subtleties could be added/addressed in the abstract or intro. If needed. But the title doesn't seem like the right place to be differentiating a JWT claim set (which is JSON) or JSON or JSON-LD or whatever.

I'm in support of this.

> title = "Verifiable Credentials with a JWT Claim Set with or without Selective Disclosure"

Author: ```suggestion
title = "JWT and SD-JWT Verifiable Credentials"
```

abbrev = "JWT VC"

> ipr = "trust2009

Author: ```suggestion
ipr = "trust200902"
```

.# Abstract

This specification describes data formats as well as validation and processing

> rules to express Verifiable Credentials with a JWT Claim set [@!RFC7519] that are protected
> using the JWS [@!RFC7515], or the SD-JWT [@!I-D.ietf-oauth-selective-disclosure-jwt]

Reviewer 2: ```suggestion
rules to express Verifiable Credentials as a JWT [@!RFC7519] or an SD-JWT [@!I-D.ietf-oauth-selective-disclosure-jwt]
```

proposed simplification

Author: I think we need to highlight that we are proposing two different ways to protect JSON payload (JWT claim set) with and without SD, and the current text is more explicit about it.

Reviewer: I think the suggested text from @tplooker conveys it too.

Author: as you said yourself, Brian, `a JWS with a JWT claim set is a JWT`, which is why I am ok calling it a JWT VC, but at least in the intro I think we agreed to make it clear that `a JWS with a JWT claim set is a JWT`

Reviewer 3: I think Tobias' proposal is just fine.

Author: ```suggestion
rules to express Verifiable Credentials based on a JWT [@!RFC7519] or an SD-JWT [@!I-D.ietf-oauth-selective-disclosure-jwt]
```

involved, a Status Provider, who delivers revocation information to Verifiers.
(The Verifier can also serve as the Status Provider.)

This specification defines Verifiable Credentials based on JWT and SD-JWT.

## Rationale

JSON Web Tokens (JWTs) [@!RFC7519] can be used to express
Verifiable Credentials in the three-party-model
in a way that is easy to understand and process as it
builds upon established web primitives.

> JWT-based credentials enable selective

Reviewer 4: ```suggestion
```

> JWT-based credentials enable selective

Reviewer 4: I suggest to remove this paragraph. It is not required in this draft.

Reviewer 2: Agreed I think we can remove this.

Author: removed

> JWT-based credentials enable selective
> disclosure, i.e., the ability for a Holder to disclose only a subset of the contained
> claims, in an Identity Provider ecosystem by issuing new JWTs to the Verifier for

Author: ```suggestion
```

> every presentation.

Reviewer 4: ```suggestion
```

> every presentation.

Author: ```suggestion
```

involved, a Status Provider, who delivers revocation information to Verifiers.
(The Verifier can also serve as the Status Provider.)

> This specification defines Verifiable Credentials based on the JWS and SD-JWT
> with a JWT Claim Set.

Reviewer 2: ```suggestion
This specification defines Verifiable Credentials based on the JWT and SD-JWT.
```

Author: same comment as above two comments..

Reviewer 3: Tobias' proposal, but remove `the`.
```suggestion
This specification defines Verifiable Credentials based on JWT and SD-JWT.
```

# Scope

* This specification defines

>   - Data model and media types for Verifiable Credentials based on JWS and SD-JWTs.

Reviewer 3: ```suggestion
  - Data model and media types for Verifiable Credentials based on JWTs and SD-JWTs.
```

TBD: conventional crypt, hardware security, hsm, mobile secure area,
compliance with FIPS

# Common Data Format for JWT VC and SD-JWT VC

JWT VCs and SD-JWT VCs MUST follow the common rules defined in this section.

## Header Parameters

This section defines JOSE header parameters for the JWT VC
and the SD-JWT VC.

The `typ` header parameter MUST be present. The `typ` value MUST

> use the values defined in (#media-type-jwt-vc) or (#media-type-sd-jwt-vc).

Reviewer 4: Should we consider to use the same `typ` header value for both variants, JWT and SD-JWT? I came across this when I was thinking about the OID4VC credential format profile for JWT-based VCs. A single `typ` would also result in a single format identifier in the presentation request, which might everyone's live easier. The only reason to distinguish would be that a dedicated format identifier for sd-jwt would allow the processor to directly decide whether selective disclosure is allowed in the presentation definition.

Reviewer 5: I'm fine starting with distinct "typ" media types, as it provides recipients an indication of whether the VC is a plain JWT or SD-JWT.  We could later decide to make that distinction using "cty" if we decide to go in that direction.

Author: it's not just the presentation request that needs to differentiate whether there is a selective disclosure or not. how would issuer, verifier, wallets indicate that they support only sd-jwt vc in their metadata? processing of sd-jwt vc and jwt vc are different enough to have separate `typ` values.

@danielfett @bc-pi what do you think?

Reviewer 4: create an issue

Reviewer: I can't help but wonder if it'd make sense to have the spec be only about SD-JWT VCs and treat the plain JWT VC as just a special case of an SD-JWT with no selectively disclosable claims.

Author: that's what @tlodderstedt has suggested, too. would that change the structure in this PR quite a lot? since the text will be all about sd-jwt vc and there will be one section saying that to do sd-jwt vc, one only need to not use `_sd` and no disclosures?

Reviewer: Yeah, I think it would change this PR quite a lot. Or a new one... Sorry!* I'm honestly not 100% sure about the idea but my intuition is that it would be a cleaner/better/simpler approach. 

* I did mention "Isn't a JWT VC just a SD-JWT VC without any selectively disclosable claims... ?" last week in some slack conversation though so this shouldn't be a total surprise.

TBD: conventional crypt, hardware security, hsm, mobile secure area,
compliance with FIPS

# Common Data Format for JWT VC and SD-JWT VC

JWT VCs and SD-JWT VCs MUST follow the common rules defined in this section.

## Header Parameters

This section defines JWT header parameters for the JWT VC

> and the SD-JWT component of the SD-JWT VC.

Author: ```suggestion
and the SD-JWT VC.
```

<{{examples/01/combined_issuance.txt}}

# Verification and Processing {#vc-sd-jwt-verification-and-processing}

> The recipient of the JWT VC MUST process and verify a JWT VC as

Author: ```suggestion
The recipient of the JWT VC MUST process and verify a JWT VC or an SD-JWT VC as
```

<{{examples/01/combined_issuance.txt}}

# Verification and Processing {#vc-sd-jwt-verification-and-processing}

The recipient of the JWT VC MUST process and verify a JWT VC as
follows:

 1. REQUIRED. To process and verify the JWT VC, follow the stepd defined in

> Section 7.2 of [@!RFC7519]. To process and verify the SD-JWT VC, follow the steps

Author: i am not a fan of `if a VC is an SD-JWT VC`. SD-JWT VC is a VC.

> Section 7.2 of [@!RFC7519]. To process and verify the SD-JWT VC, follow the steps

Reviewer 4: but not every VC is a SD-JWT

> Section 7.2 of [@!RFC7519]. To process and verify the SD-JWT VC, follow the steps

Reviewer 6: Actually every VC (JSON or JSON-LD) can be protected with SD-JWT
same as every JSON or JSON-LD can be protected with JWS
